packages/json-schema/test/extension.test.ts

```diff
@@ -0,0 +1,232 @@
+      @extension("x-record", Record<{ name: string }>)
+      model Foo {
+        @extension("x-bool", boolean)
+        @extension("x-bool-literal", typeof true)
+        @extension("x-int", int8)
+        @extension("x-int-literal", typeof 42)
+        @extension("x-string", string)
+        @extension("x-string-literal", typeof "hi")
+        @extension("x-union", "one" | "two")
+        @extension("x-null", typeof null)
+        x: string;
+      }
+
+      model Collection<Item> {
+        size: int32;
+        item: Item[];
+      }
+      
+      model Thing {
+        name: string;
+      }
+    `);
+  const Foo = schemas["Foo.json"];
+
+  assert.deepStrictEqual(Foo["x-named-model"], { $ref: "Thing.json" });
+  assert.deepStrictEqual(Foo["x-model-template"], { $ref: "CollectionThing.json" });
+  assert.deepStrictEqual(Foo["x-record"], {
+    additionalProperties: {
+      properties: {
+        name: {
+          type: "string",
+        },
+      },
+      required: ["name"],
+      type: "object",
+    },
+    properties: {},
+    type: "object",
+  });
+  assert.deepStrictEqual(Foo.properties.x["x-bool"], { type: "boolean" });
+  assert.deepStrictEqual(Foo.properties.x["x-bool-literal"], {
+    const: true,
+    type: "boolean",
+  });
+  assert.deepStrictEqual(Foo.properties.x["x-int"], {
+    minimum: -128,
+    maximum: 127,
+    type: "integer",
+  });
+  assert.deepStrictEqual(Foo.properties.x["x-int-literal"], {
+    const: 42,
+    type: "number",
+  });
+  assert.deepStrictEqual(Foo.properties.x["x-string"], {
+    type: "string",
+  });
+  assert.deepStrictEqual(Foo.properties.x["x-string-literal"], {
+    const: "hi",
+    type: "string",
+  });
+  assert.deepStrictEqual(Foo.properties.x["x-union"], {
+    anyOf: [
+      { const: "one", type: "string" },
+      { const: "two", type: "string" },
+    ],
+  });
+  assert.deepStrictEqual(Foo.properties.x["x-null"], {
+    type: "null",
+  });
+});
+
+it("handles Json-wrapped types", async () => {
+  const schemas = await emitSchema(`
+      @extension("x-anon-model", Json<{ name: "foo" }>)
+      @extension("x-named-model", Json<Thing>)
+      @extension("x-nested-anon-model", Json<{ items: [ {foo: "bar" }]}>)
+      model Foo {
+        @extension("x-bool-literal", Json<true>)
+        @extension("x-int-literal", Json<42>)
+        @extension("x-string-literal", Json<"hi">)
+        x: string;
+      }
+
+      model Collection<Item> {
+        size: 1;
+        item: Item[];
+      }
+      
+      model Thing {
+        name: "thing";
+      }
+    `);
+  const Foo = schemas["Foo.json"];
+
+  assert.deepStrictEqual(Foo["x-anon-model"], { name: "foo" });
+  assert.deepStrictEqual(Foo["x-named-model"], { name: "thing" });
+  assert.deepStrictEqual(Foo["x-nested-anon-model"], { items: [{ foo: "bar" }] });
+
+  // Would expect this to be `{ size: 1, item: { name: "thing" }}` but it isn't.
+  // assert.deepStrictEqual(Foo["x-model-template"], { size: 1, item: {} });
```

Comment:
Say more?

Reply:
Thinking about it more - I think I understand what's happening. Taking this as an example:
```
@extension("x-example", Json<Collection<{ name: "foo" }>>)
```
[playground link using generic array](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvanNvbi1zY2hlbWEiOwoKdXNpbmcgVHlwZVNwZWMuSnNvblPFHTsKCm5hbWVzcGFjZSDGE3M7CgoKQMRBxhAKbW9kZWwgRm9vIHsKICBAZXh0ZW5zaW9uKCJ4LWV4YW1wbGUiLCDEVDxDb2xsZWN0aW9uPHsgxFk6ICJmb28iIH0%2BPikKICB4OiBzdHJpbmc7Cn3HWMszSXRlbT7FZXNpemU6IDE7CiAgaXRlbTogxBtbXcQ2xzdUaOQA0sQsx2d0xBEixCI%3D&e=%40typespec%2Fjson-schema&options=%7B%7D)

I was expecting to see this output:
```
x-example:
  size: 1
  item:
    - name: foo
```
and indeed, if I change Collection to use a tuple instead of an array, it does work that way:
[playground link using generic in tuple](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvanNvbi1zY2hlbWEiOwoKdXNpbmcgVHlwZVNwZWMuSnNvblPFHTsKCm5hbWVzcGFjZSDGE3M7CgoKQMRBxhAKbW9kZWwgRm9vIHsKICBAZXh0ZW5zaW9uKCJ4LWV4YW1wbGUiLCDEVDxDb2xsZWN0aW9uPHsgxFk6ICJmb28iIH0%2BPikKICB4OiBzdHJpbmc7Cn3HWMszSXRlbT7FZXNpemU6IDE7CiAgaXRlbTogW8QcXcQ2xzdUaOQA0sQsx2d0xBEixCI%3D&e=%40typespec%2Fjson-schema&options=%7B%7D)

I think requiring a tuple over an array makes sense here, since I'm meant to be passing in a value - not a type - when using the `Json` model. That said - I was surprised it generated an object value in the output instead of giving me a warning.